File: blood-bank-be/bloodBank/src/main/java/ftn/uns/ac/rs/bloodbank/registration/dto/CustomerRequest.java
package ftn.uns.ac.rs.bloodbank.registration.dto;

import ftn.uns.ac.rs.bloodbank.applicationUser.model.UserRole;
import ftn.uns.ac.rs.bloodbank.sharedModel.GenderType;
import lombok.*;
@Data
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Setter

public class CustomerRequest {
    @NonNull
    private String username;
    @NonNull
    private String password;
    @NonNull
    private String name;
    @NonNull
    private String surname;
    @NonNull
    private String phone;
    @NonNull
    private String jmbg;
    @NonNull
    private String email;
    @NonNull
    private AddressRequest address;
    @NonNull
    private UserRole role;
    @NonNull
    private ProfessionRequest profession;
    private GenderType gender;



}

File: blood-bank-fe/src/app/model/CustomerRequest.ts
import {AddressRequest} from "./AddressRequest";
import {ProfessionRequest} from "./ProfessionRequest";

export interface ICustomerRequest {
  name?: string;
  username?: string;
  password?: string;
  surname?: string,
  phone?: string,
  jmbg?: string,
  email?: string,
  role?: string,
  address?: AddressRequest,
  profession? : ProfessionRequest
}
export class CustomerRequest implements ICustomerRequest {
  address?: AddressRequest;
  email?: string;
  jmbg?: string;
  name?: string;
  password?: string;
  phone?: string;
  surname?: string;
  role?: string;
  username?: string;
  profession? : ProfessionRequest;
  constructor(data?: ICustomerRequest){
    if (data) {
      for (const property in data) {
        if (data.hasOwnProperty(property))
          (<any>this)[property] = (<any>data)[property];
      }
    }
  }
}
